### ISSUE TITLE
JAX-RS integration and webPathPrefix

### ISSUE THREAD

**Issue by lodrantl** [crnk-project/crnk-framework#707]
I have a JAX-RS service (on Wildfly) and am trying to include Crnk. It has an ApplicationPath of /myapp, but I want Crnk to reside at /myapp/api.

When you set the `webPrefixPath` using `feature.getBoot().setWebPathPrefix("/api");` nothing happens.

In the CrnkFeature.java we have:
```
    public String getWebPathPrefix() {
        return UrlUtils.removeLeadingSlash(boot.getWebPathPrefix());
    }
```
which is used in JaxrsRequestContext.java:
```
	private String buildPath(UriInfo uriInfo) {
		String basePath = uriInfo.getPath();
		String webPathPrefix = feature.getWebPathPrefix();
		String path;
		if (webPathPrefix != null && basePath.startsWith(webPathPrefix)) {
			path = basePath.substring(webPathPrefix.length());
		} else {
			path = basePath;
		}
		return path == null || path.isEmpty() ? "/" : path;
	}
```
There are two problems here:

1. `basePath.startsWith(webPathPrefix)` is never true because basePath starts with a slash which is explicitly removed from webPathPrefix.
2. Even if you keep the leading slash that would mean the Crnk repositories are served both without prefix and with prefix which I do not think is correct.

Am I doing something wrong or is this an actual problem of the implementation.
Thanks for your help guys.

**Comment by remmeier** [comment]
unfortunately different frameworks work a bit differently in the area. So at some point we started just ignoring preceding slashes. But not completely as it seems. Things should work well if you drop the prefix?`meaning just `api`  rather than `/api`? But this should change.

**Comment by lodrantl** [comment]
No, the problem is that basePath received from uriInfo.getPath() contains a leading slash, while the webPathPrefix will never contain it.

Problem 2 still remains, would modifying CrnkFilter to only pass requests matching the webpathPrefix work?

And also I found a third problem. The basePath returned at
https://github.com/crnk-project/crnk-framework/blob/44718c0edc98d0fb20af35fbd46cb071003b04f8/crnk-setup/crnk-setup-rs/src/main/java/io/crnk/rs/JaxrsRequestContext.java#L77
does not contain the webpathPrefix so the urls in links point to the wrong targets. I can submit a PR with my proposed changes if you want.

**Comment by remmeier** [comment]
PR would be great. Given that one has to be careful with different frameworks, we should focus on changes to `JaxrsRequestContext`, but try to avoid other ones.

**Comment by remmeier** [comment]
I will keep the ticket open to clean up the area a bit to make it more robust. Probably best by moving away from basic strings.
